brite2caom2/tests/test_preview_augmentation.py

```diff
@@ -0,0 +1,126 @@
+#
+#  OpenCADC is distributed in the       OpenCADC est distribué
+#  hope that it will be useful,         dans l’espoir qu’il vous
+#  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+#  without even the implied             GARANTIE : sans même la garantie
+#  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+#  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+#  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+#  General Public License for           Générale Publique GNU Affero
+#  more details.                        pour plus de détails.
+#
+#  You should have received             Vous devriez avoir reçu une
+#  a copy of the GNU Affero             copie de la Licence Générale
+#  General Public License along         Publique GNU Affero avec
+#  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+#  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+#                                       <http://www.gnu.org/licenses/>.
+#
+#  : 4 $
+#
+# ***********************************************************************
+#
+
+import glob
+from os.path import basename
+
+from caom2pipe.caom_composable import get_all_artifact_keys
+from caom2pipe import manage_composable as mc
+
+import brite2caom2.storage_name
+from brite2caom2 import preview_augmentation, main_app, reader
+from mock import Mock, patch
+import test_main_app
+
+
+def pytest_generate_tests(metafunc):
+    obs_id_list = [ii for ii in glob.glob(f'{test_main_app.TEST_DATA_DIR}/*') if '.expected.xml' in ii]
+    metafunc.parametrize('test_name', obs_id_list)
+
+
+@patch('caom2pipe.client_composable.ClientCollection')
+def test_preview_visit(clients_mock, test_name):
+    original_collection = mc.StorageName.collection
+    try:
+        obs_id = basename(test_name).replace('.expected.xml', '')
+        dir_name = obs_id.split('_')[0]
+        rlog_fqn = f'{test_main_app.TEST_DATA_DIR}/{dir_name}/{obs_id}.rlogdb'
+        orig_fqn = f'{test_main_app.TEST_DATA_DIR}/{dir_name}/{obs_id}.orig'
+        for test_fqn in [orig_fqn, rlog_fqn]:
+            mc.StorageName.collection = main_app.COLLECTION
+            metadata_reader = reader.BriteFileMetadataReader()
+            test_obs = mc.read_obs_from_file(test_name)
+            # pre-condition
+            uris = get_all_artifact_keys(test_obs)
+            assert len(uris) == 5, f'precondition failure {test_name}'
+            test_storage_name = brite2caom2.storage_name.BriteName(test_fqn)
+            metadata_reader.set(test_storage_name)
+            kwargs = {
+                'working_directory': test_main_app.TEST_DATA_DIR,
+                'cadc_client': None,
+                'metadata_reader': metadata_reader,
+                'observable': Mock(),
+                'storage_name': test_storage_name,
+            }
+            test_obs = preview_augmentation.visit(test_obs, **kwargs)
```

Comment:
Not sure why `kwargs`. It should be sufficient to say `visit(test_obs, working_directory=test_main_app.TEST_DATA_DIR,...)`

Reply:
Is that the more Pythonic way to handle it?